### PR TITLE
DR-3406: Send slack alerts on nightly test failures; Update to use slack's GHA Integration

### PR DESCRIPTION
**Pull request by snf2ye**
https://broadworkbench.atlassian.net/browse/DR-3406

Remaining TODO:

- [x] Get SLACK_BOT_TOKEN added to GHA secrets via [terraform-ap-deployments PR](https://github.com/broadinstitute/terraform-ap-deployments/pull/1323/files)
- [x] Rename slack bot token secret in PR
- [x] Remove TEST_SLACK_BOT_TOKEN secret
- [ ] Once PR is merged, remove webhook url secret

___________

Example Output: 
Nightly runs of e2e tests report to #dsde-qa.
Successful runs report to #jade-spam:
image

Failures report to #jade-alerts:
image

Following a similar pattern to what we did in [DRS Hub](https://github.com/DataBiosphere/terra-drs-hub/pull/100). 
### Using Slack GHA integration with Slack Apps
- Utilizing the [slack-backed GHA](https://github.com/slackapi/slack-github-action?tab=readme-ov-file#technique-2-slack-app) integration
- Created new slack app named "Data Repo UI Notification" and added app to #jade-alerts, #jade-spam and #dsde-qa channel
- Gave the bot associated with the app "chat:write" permissions and then added the bot's token to github action secrets
- App can be managed here: https://api.slack.com/apps/A06C4JEBDTR

### CODE REVIEW

**Review by nmalfroy — APPROVED** [review]
this looks great! love the image :)

File: .github/workflows/dev-image-update.yaml [hunk 1]
```diff
@@ -98,19 +98,10 @@ jobs:
...
-        author_name: Integration Test
-      env:
-        GITHUB_TOKEN: ${{ secrets.BROADBOT_TOKEN }}
-        SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+  notify-slack:
+    needs: [ update_image, helm_tag_bump, cherry_pick_image_to_production_gcr ]
+    uses: ./.github/workflows/notify-slack.yaml
+    secrets: inherit
+    with:
+      workflow_name: Dev Image Udpate
```
Comment (okotsopoulos) [inline .github/workflows/dev-image-update.yaml]: Small typo :)

```suggestion
      workflow_name: Dev Image Update
```

**Review by okotsopoulos — APPROVED** [review]

File: .github/workflows/dev-image-update.yaml (hunk 1 above)
Comment (nmalfroy) [reply .github/workflows/dev-image-update.yaml]: wow...nice catch.  It took me a minute in my email to spot the difference

Comment (snf2ye) [reply .github/workflows/dev-image-update.yaml]: Oh gosh, thank you!

File: .github/workflows/dev-image-update.yaml [hunk 2]
```diff
@@ -98,19 +98,10 @@ jobs:
...
-    - name: Slack job status
-      uses: broadinstitute/action-slack@v3.15.0
-      with:
-        status: ${{ contains(needs.*.result, 'failure') && 'failure' || 'success' }}
-        author_name: Integration Test
-      env:
-        GITHUB_TOKEN: ${{ secrets.BROADBOT_TOKEN }}
-        SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+  notify-slack:
+    needs: [ update_image, helm_tag_bump, cherry_pick_image_to_production_gcr ]
```
Comment (pshapiro4broad) [inline .github/workflows/dev-image-update.yaml:102]: Do you need to `need` all the jobs here? If you need `cherry_pick_image_to_production_gcr`, it transitively depends on the other things.
```suggestion
    needs: [ cherry_pick_image_to_production_gcr ]
```

Comment (snf2ye) [reply .github/workflows/dev-image-update.yaml:102]: It seems like you wouldn't, but the examples in the documentation indicate that you should still list them all: 
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-requiring-successful-dependent-jobs

**Pull request merged**